pkg/controller/sparkapplication/controller.go

```diff
@@ -126,21 +129,41 @@ func newSparkApplicationController(
 		controller.metrics.registerMetrics()
 	}
 
+	enableNamespaceFilter := util.IsNamespaceFilterEnabled(namespaceFilterConfig)
+	var namespaceFilter *regexp.Regexp
+	if enableNamespaceFilter {
+		namespaceFilter = util.GetNamespaceFilter(namespaceFilterConfig.NamespaceFilter)
+	}
+
 	crdInformer := crdInformerFactory.Sparkoperator().V1beta2().SparkApplications()
-	crdInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+	var resourceEventHandlerFuncs cache.ResourceEventHandler = cache.ResourceEventHandlerFuncs{
 		AddFunc:    controller.onAdd,
 		UpdateFunc: controller.onUpdate,
 		DeleteFunc: controller.onDelete,
-	})
+	}
+	if enableNamespaceFilter {
+		resourceEventHandlerFuncs = cache.FilteringResourceEventHandler{
+			FilterFunc: filterSparkAndPodEventsByNamespace(*namespaceFilter),
+			Handler: resourceEventHandlerFuncs,
```

Comment:
strange situation: resourceEventHandlerFuncs depends on resourceEventHandlerFuncs. (I know it is a copy of resourceEventHandlerFuncs passed to cache.FilteringResourceEventHandler{... Handler: ...} )
confused when reading, but the code is good

Reply:
Changed and enhanced with respect to our discussion of yesterday